src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -88,25 +88,32 @@ public abstract class RegexRunner
         protected Match? Scan(Regex regex, string text, int textbeg, int textend, int textstart, int prevlen, bool quick) =>
             Scan(regex, text, textbeg, textend, textstart, prevlen, quick, regex.MatchTimeout);
 
-        protected internal Match? Scan(Regex regex, string text, int textbeg, int textend, int textstart, int prevlen, bool quick, TimeSpan timeout)
+        protected internal virtual void Scan(ReadOnlySpan<char> text)
         {
-            this.quick = quick;
+            string? s = runtext;
+            // If beginning was passed in, then runtext and span won't match lengths so calculate the beginning to be used for the comparison.
+            int beginning = (s != null && s.Length != text.Length) ? s.Length - text.Length : 0;
+            if (s == null || text != s.AsSpan(beginning, text.Length))
```

Comment:
This is clever. I hadn't comprehended what you were doing here before, but you're basically assuming (validly) that the span will always be a subset of the string, and you're using that to recreate the original runtextbeg and runtextend values without complicating the 99.99% case to store them earlier. Nice. A more detailed comment about that would be helpful.

Reply:
yeah the idea was to not require to pass in runtextbeg and runtextend everytime but now that I look at it again, it won't work in all cases so I think I'll have to rewrite this. The scenario where this won't work is if I do something like:

```c#
myPrecompiledRegex.Match("Some string that has many words but I only want to test the second one", beginning: 5, length: 6);
```

the slice would only be the word `string`, but when I calculate beginning I will be incorrectly setting it to be the last 6 characters `nd one`.  I can see two possible solutions to this, one is to pass in runtextbeg every time, or the other is to special case precompiled scenario (which is the only one that cares about original runtextbeg and end) where it calls back into a new internal property in RegexRunner where we previously save the value during Run().